model/task/task.go

```diff
@@ -658,25 +660,30 @@ func GenerateNotRun() ([]Task, error) {
 		StartTimeKey:      bson.M{"$gt": time.Now().Add(-maxGenerateTimeAgo)}, // ignore older tasks, just in case
 		GenerateTaskKey:   true,                                               // task contains generate.tasks command
 		GeneratedTasksKey: bson.M{"$exists": false},                           // generate.tasks has not yet run
-		GeneratedJSONKey:  bson.M{"$exists": true},                            // config has been posted by generate.tasks command
+		"$or": []bson.M{
+			bson.M{GeneratedJSONAsStringKey: bson.M{"$exists": true}}, // config has been posted by generate.tasks command
+			bson.M{GeneratedJSONKey: bson.M{"$exists": true}},         // TODO Remove after EVG-6759 is deployed
+		},
 	}))
 }
 
 // SetGeneratedJSON sets JSON data to generate tasks from.
 func (t *Task) SetGeneratedJSON(json []json.RawMessage) error {
-	if len(t.GeneratedJSON) > 0 {
+	if len(t.GeneratedJSONAsString) > 0 || len(t.GeneratedJSON) > 0 {
 		return nil
 	}
-	t.GeneratedJSON = json
+	s := []string{}
+	for _, j := range json {
+		s = append(s, string(j))
+	}
+	t.GeneratedJSONAsString = s
 	return UpdateOne(
 		bson.M{
 			IdKey: t.Id,
-			// If this field already is set, something has gone wrong.
-			GeneratedJSONKey: bson.M{"$exists": false},
 		},
 		bson.M{
 			"$set": bson.M{
-				GeneratedJSONKey: json,
```

Comment:
Should we still set GeneratedJSONKey for now in case of a revert?

Reply:
Yeah, fair. Fixed.

Reply:
I think for this to have any effect you need to change the bson tags? It looks like right now they're both being stored as `generate_json`, idk if that's a problem.

Reply:
They actually differ. One is "generated", since I realized that's actually the name I wanted, not "generate".